.github/workflows/typecheck.yml

```diff
@@ -30,7 +30,7 @@ jobs:
           #    - git diff is used to see the files that were added on this branch
           #    - gh pr view is used to list files touched by this PR. Git diff may give false positives if the branch isn't up-to-date with main
           #  - wc counts the words in the result of the intersection
-          count_new_js=$(comm -1 -2 <(git diff --name-only --diff-filter=A origin/main HEAD -- 'src/*.js') <(gh pr view ${{ github.event.pull_request.number }} --json files | jq -r '.files | map(.path) | .[]') | wc -l)
+          count_new_js=$(comm -1 -2 <(git diff --name-only --diff-filter=A origin/main HEAD -- 'src/*.js' '__mocks__/*.js' '.github/*.js' '.storybook/*.js' 'assets/*.js' 'config/*.js' 'desktop/*.js' 'jest/*.js' 'scripts/*.js' 'tests/*.js' 'web/*.js' 'workflow_tests/*.js') <(gh pr view ${{ github.event.pull_request.number }} --json files | jq -r '.files | map(.path) | .[]') | wc -l)
```

Comment:
`'.github/*.js'` I'm a bit worried about this one as we generate index.js files that are committed to the repo, if we create a new one Typescript check will fail